internal/disk/customizations.go

```diff
@@ -0,0 +1,104 @@
+package disk
+
+import (
+	"io"
+	"math/rand"
+
+	"github.com/google/uuid"
+	"github.com/osbuild/osbuild-composer/internal/blueprint"
+	"github.com/osbuild/osbuild-composer/internal/distro"
+)
+
+const (
+	sectorSize            = 512
+	BIOSBootPartitionGUID = "21686148-6449-6E6F-744E-656564454649"
+	BIOSBootPartitionUUID = "FAC7F1FB-3E8D-4137-A512-961DE09A5549"
+
+	FilesystemDataGUID = "0FC63DAF-8483-4772-8E79-3D69D8477DE4"
+	FilesystemDataUUID = "CB07C243-BC44-4717-853E-28852021225B"
+
+	EFISystemPartitionGUID = "C12A7328-F81F-11D2-BA4B-00A0C93EC93B"
+	EFISystemPartitionUUID = "68B2905B-DF3E-4FB3-80FA-49D1E773AA33"
+	EFIFilesystemUUID      = "7B77-95E7"
+
+	RootPartitionUUID = "6264D520-3FB9-423F-8AB8-7A0A8E3D3562"
+)
+
+func CreatePartitionTable(
+	mountpoints []blueprint.FilesystemCustomization,
+	imageSize uint64,
+	archName string,
+	basePartitionTable PartitionTable,
+	rng *rand.Rand,
+) PartitionTable {
+
+	basePartitionTable.Size = imageSize
+	partitions := []Partition{}
+
+	if bootPartition := basePartitionTable.BootPartition(); bootPartition != nil {
+		// the boot partition UUID needs to be set since this
+		// needs to be randomly generated
+		bootPartition.Filesystem.UUID = uuid.Must(newRandomUUIDFromReader(rng)).String()
+	}
+
+	// start point for all of the arches is
+	// 2048 sectors.
+	var start uint64 = basePartitionTable.updatePartitionStartPointOffsets(2048)
+
+	for _, m := range mountpoints {
+		if m.Mountpoint != "/" {
+			partitionSize := uint64(m.MinSize) / sectorSize
+			partition := createPartition(m.Mountpoint, partitionSize, start, archName, rng)
+			partitions = append(partitions, partition)
+			start += uint64(m.MinSize / sectorSize)
+		}
+	}
+
+	// treat the root partition as a special case
+	// by setting the size dynamically
+	rootPartition := basePartitionTable.RootPartition()
+	rootPartition.Start = start
+	rootPartition.Size = ((imageSize / sectorSize) - start - 100)
```

Comment:
You're right, it doesn't actually do a check. I can add a check and a test to cover this.

However, the image size for all the partitions has been taken into account, 
his was set in a previous pr. We add up all the partition sizes and and update the size here: 
https://github.com/osbuild/osbuild-composer/blob/42480f7b265e8619dfb42fc0f6aa246e5e9d3421/internal/weldr/api.go#L2283

Reply:
Actually, since the custom sizes are `minimum` sizes, maybe we can increase the partition table size rather than throwing an error, @teg what are your thoughts?

Reply:
Thanks for pointing this out. Nevertheless it may be still a good idea to make sure that we don't create buggy partitioning table even though there is currently no code path when this could happen. This may become an issue once we allow customising the filesystem via other APIs.

Reply:
I agree, increase to make things fit. However, if this is handle elsewhere, so the code path would never be taken, throwing an error for now might make the most sense.